### PR TITLE
[WIP] Implement cycle-restart support

### PR DESCRIPTION
**Pull request by Widdershin**
Please don't merge!

There is still lots to do, I just wanted to open this so we could keep track of what there is to do and start making travis happy.

This is to add compatibility for cycle-restart, to enable the logging and replay of application IO so that we can hot load code and do real time-traveling.

### CODE REVIEW

**Comment by staltz** [comment]
Cool!

One question concerning any driver: how easy will it be for one-liner drivers to support cycle-restart?

File: src/http-driver.js [hunk 1]
```diff
@@ -123,24 +124,64 @@ function isolateSource(response$$, scope) {
...
       .replay(null, 1)
+
     response$$.connect()
     response$$.isolateSource = isolateSource
     response$$.isolateSink = isolateSink
+    response$$.history = () => history
     return response$$
   }
+
+  httpDriver.aboutToReplay = () => {
```
Comment (staltz) [inline src/http-driver.js]: Since these seem to be lifecycle hooks, I'd suggest the name `onPreReplay`

File: src/http-driver.js [hunk 2]
```diff
@@ -123,24 +124,64 @@ function isolateSource(response$$, scope) {
...
+    replaying = true
+  }
+
+  httpDriver.replayFinished = () => {
```
Comment (staltz) [inline src/http-driver.js]: `onPostReplay` ?

**Comment by Widdershin** [comment]
I think it will be possible to create a function that will take in a driver and return that same driver with logging and replay implemented, assuming that the driver takes in and returns a single stream.

**Comment by Widdershin** [comment]
Closed as the new implementation of cycle-restart provides a function to implement logging and replay on drivers.

https://github.com/Widdershin/cycle-restart/

**Pull request closed without merging**